### PR TITLE
DCGAN on OASIS brain images - Amy Zhao

### PR DESCRIPTION
**Pull request by amyzhao11**
I implemented a DCGAN on the OASIS brain MRI images

Note: I had previously been committing my larger ipynb files and came across an issue where git would not allow me to push my local repo as the files were too big. Deleting and committing these changes did not work as previous versions of this large file still existed in prior commits. I went to a prac session today (06/11) to get help with this and we could not find a solution to this so I have sent a copy of my git log to Scott (the tutor) as proof of my commits. I also have a copy of this if you would like proof. Because of this issue, I had to copy my current files to a new branch in order to push and then do a pull request so it is missing all the commits that I did prior to today. I also only included a driver script, model script and resource folder (containing sample images) in this pull request.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Commit messages should be more specific - will leave this up to Scott
- Incorrect source and target branches. You should be using and targeting the `topic-recognition` branch
- Code valid for Hard Difficulty
- Results are good

Other Feedback
- Please follow python file and folder naming convention e.g. `lorem_ipsum_123_abc.py`

**Comment by rexsw** [comment]
Code looks good for hard difficulty

Commit messages from before the git issue are good (I have a copy if you want to see them)

For next time;

Rather than making use of a whole library for one function, it is better to simply write that function yourself, that way your project doesn’t also need skimage 

Comments such as 
    #Define loss functions
    cross_entropy = tf.keras.losses.BinaryCrossentropy(from_logits=True)
Add nothing to the readability of the code and should be avoided. Both overall comments are good. 

Code in files such a driverscript.py should be in a main function

**Comment by shakes76** [comment]
I can't identify you Amy, there are 3 students with similar names, please provide ID or first name.

**Comment by amyzhao11** [comment]
Hi Shakes, my student number is 43571806 and name Amy Zhao

**Comment by shakes76** [comment]
Thanks. Sorry my mistake navigating the grade center. :(

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Please switch to the correct branch topic-recognition.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by shakes76** [comment]
Re-lodged as #188 to correct branch, closing this one.
